src/openrct2/rct2/S6Importer.cpp

```diff
@@ -262,11 +262,22 @@ class S6Importer final : public IParkImporter
 
         gResearchPriorities = _s6.active_research_types;
         gResearchProgressStage = _s6.research_progress_stage;
-        gResearchLastItem.rawValue = _s6.last_researched_item_subject;
+        if (_s6.last_researched_item_subject != RCT12_RESEARCHED_ITEMS_SEPARATOR)
+            gResearchLastItem->rawValue = _s6.last_researched_item_subject;
+        else
+            gResearchLastItem = std::nullopt;
         // pad_01357CF8
-        gResearchNextItem.rawValue = _s6.next_research_item;
+        if (_s6.next_research_item != RCT12_RESEARCHED_ITEMS_SEPARATOR)
+        {
+            gResearchNextItem->rawValue = _s6.next_research_item;
+            gResearchNextItem->category = _s6.next_research_category;
```

Comment:
Setting the fields directly on an optional does not change the has_value flag. Make sure you use something like `std::make_optional` first otherwise `gResearchNextItem` will still be `std::nullopt`.

Reply:
Do I need to use `std::make_optional` on all assignments, or just on those where I set fields directly?

Reply:
`std::make_optional` will give you the constructor of the type. If you have a constructor that lets you set all the fields, then you can just make one single call to `std::make_optional`. If you want to set fields 1 by 1, then you need to call `std::make_optional` to reset it to the default initial value of an initialised `ResearchItem` then set the fields.